pkg/xray/reports.go

```diff
@@ -0,0 +1,802 @@
+package xray
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+var getReportSchema = func(filtersSchema map[string]*schema.Schema) map[string]*schema.Schema {
+	return util.MergeMaps(
+		getProjectKeySchema(false, ""),
+		map[string]*schema.Schema{
+			"report_id": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				Computed:         true,
+				Description:      "Report ID",
+				ValidateDiagFunc: validator.StringIsNotEmpty,
```

Comment:
Do we need validation func as this is a computed attribute and it can't be set by users?

Reply:
Yeah, we don't. Removed.